test/react_context_test.dart

```diff
@@ -0,0 +1,159 @@
+@TestOn('browser')
+@JS()
+library react_test_utils_test;
+
+import 'dart:html' as html;
+
+import 'package:js/js.dart';
+import 'package:test/test.dart';
+
+import 'package:react/react.dart' as react;
+import 'package:react/react_client.dart';
+import 'package:react/react_dom.dart' as react_dom;
+
+import 'shared_type_tester.dart';
+
+main() {
+  setClientConfiguration();
+
+  testTypeValue(dynamic typeToTest) {
+    var mountNode = new html.DivElement();
+    var contextTypeRef;
+    var consumerRef;
+    react_dom.render(
+        ContextProviderWrapper({
+          'contextToUse': TestContext,
+          'value': typeToTest,
+        }, [
+          ContextTypeComponent({
+            'ref': (ref) {
+              contextTypeRef = ref;
+            }
+          }),
+          ContextConsumerWrapper({
+            'contextToUse': TestContext,
+            'ref': (ref) {
+              consumerRef = ref;
+            }
+          })
+        ]),
+        mountNode);
+    expect(contextTypeRef.context, same(typeToTest));
+    expect(consumerRef.latestValue, same(typeToTest));
+  }
+
+  group('New Context API (Component2 only)', () {
+    group('sets and retrieves values correctly:', () {
+      sharedTypeTests(testTypeValue);
+    });
+
+    group('calculateChangeBits argument functions correctly', () {
+      var mountNode = new html.DivElement();
+      _ContextProviderWrapper providerRef;
+      _ContextConsumerWrapper consumerEvenRef;
+      _ContextConsumerWrapper consumerOddRef;
+
+      setUp(() {
+        react_dom.render(
+            ContextProviderWrapper({
+              'contextToUse': TestCalculateChangedBitsContext,
+              'mode': 'increment',
+              'ref': (ref) {
+                providerRef = ref;
+              }
+            }, [
+              ContextConsumerWrapper({
+                'key': 'EvenContextConsumer',
+                'contextToUse': TestCalculateChangedBitsContext,
+                'unstable_observedBits': 1 << 2,
+                'ref': (ref) {
+                  consumerEvenRef = ref;
+                }
+              }),
+              ContextConsumerWrapper({
+                'key': 'OddContextConsumer',
+                'contextToUse': TestCalculateChangedBitsContext,
+                'unstable_observedBits': 1 << 3,
+                'ref': (ref) {
+                  consumerOddRef = ref;
+                }
+              })
+            ]),
+            mountNode);
+      });
+      test('on first render', () {
+        expect(consumerEvenRef.latestValue, 1);
+        expect(consumerOddRef.latestValue, 1);
+      });
+      test('on value updates', () {
```

Comment:
#nit add space between tests
```suggestion

      test('on value updates', () {
```